### PR TITLE
VMO-5680 Expression validation ❗ Do NOT merge

### PR DESCRIPTION
**Pull request by object-required**
This PR attempts using a custom `format` specification to employing AJV's feature to validate FLOIP Expressions.

https://user-images.githubusercontent.com/3971957/161676060-0de099c7-f927-4db6-bcdb-c3ccf9132d78.mp4

### CODE REVIEW

**Comment by jcbashdown** [comment]
Nice! This along with the `flow-runner` change seems incredibly neat. Do you feel that anything is missing from this that will still take time/require thought in VMO-5090 or is this basically done?

**Comment by jcbashdown** [comment]
Might be useful to link to the video in the ticket here also.

File: src/store/flow/block.ts [hunk 1]
```diff
@@ -81,14 +81,21 @@ export const mutations: MutationTree<IFlowsState> = {
...
   block_updateConfigByPath(state, {blockId, path, value}: {blockId: string, path: string, value: object | string}) {
-    // todo: this might still break reactivity
-    // Make nested assignment reactive
-    setWith(
-      findBlockOnActiveFlowWith(blockId, state as unknown as IContext).config!,
-      path,
-      value,
-      (nestedValue, key, nestedObject) => Vue.set(nestedObject, key, nestedValue),
-    )
+    const base = findBlockOnActiveFlowWith(blockId, state as unknown as IContext).config
```
Comment (safydy-r) [inline src/store/flow/block.ts:84]: thanks for trying to address the TODO, wonder if this approach `cloneDeep` would be more clearer ? https://bitbucket.org/voto/voto5/src/55efeb5596050fba0c9c06c0333f64d7ecef38a9/resources/js/trees/stores/trees.js#lines-198

**Review by safydy-r — APPROVED** [review]
Approving as spike PR, left comments for vmo-5090 though

File: src/store/flow/block.ts (hunk 1 above)
Comment (safydy-r) [reply src/store/flow/block.ts:84]: @jcbashdown , fyi: To answer your question here-above, another thing that I flagged in the other [PR](https://www.youtube.com/watch?v=ihrUIPfvTh8) is that we need to consider IResourceValue interface as well.

**Pull request closed without merging**